src/main/java/io/pinecone/configs/PineconeConnection.java

```diff
@@ -139,10 +139,17 @@ private ManagedChannel buildChannel() {
         NettyChannelBuilder builder = NettyChannelBuilder.forTarget(endpoint);
 
         try {
-            builder = builder.overrideAuthority(endpoint)
-                    .negotiationType(NegotiationType.TLS)
-                    .sslContext(GrpcSslContexts.forClient().build())
-                    .userAgent(config.getUserAgent());
+            if(config.isTLSEnabled()) {
+                builder = builder
+                        .overrideAuthority(endpoint)
+                        .negotiationType(NegotiationType.TLS)
+                        .sslContext(GrpcSslContexts.forClient().build())
+                        .userAgent(config.getUserAgent());
```

Comment:
Seems like we would want to set the user agent in both cases

Reply:
when tls is disabled, how do we plan on receiving the user-agent in the logs? Or are you thinking of intercepting the call via proxy for debugging purpose? Like what are the other use cases we are looking at with TLS disabled and having the user-agent?